csrc/anemone_mempool.c

```diff
@@ -1,11 +1,12 @@
 #include "anemone_mempool.h"
 #include <string.h>
+#include <malloc.h>
 
-// Allocate a new block with given predecessor 
+// Allocate a new block with given predecessor
 ANEMONE_STATIC
 anemone_block_t * anemone_block_create (anemone_block_t *prev, size_t num_bytes)
 {
-  void     *ptr = malloc (num_bytes);
+  void     *ptr = memalign(ANEMONE_ALIGNMENT, num_bytes);
```

Comment:
turns out `malloc` should suffice here, since:

> malloc is guaranteed to return memory that's suitably aligned for storing any object that has a fundamental alignment and that could fit in the amount of memory that's allocated.